pysatMissions/instruments/missions_ephem.py

```diff
@@ -0,0 +1,244 @@
+    line1 = ''.join(('1 25544U 98067A   18135.61844383  .00002728  00000-0  ',
+                     '48567-4 0  9998'))
+    line2 = ''.join(('2 25544  51.6402 181.0633 0004018  88.8954  22.2246 ',
+                     '15.54059185113452'))
+    # Use ISS defaults if not provided by user
+    if TLE1 is not None:
+        line1 = TLE1
+    if TLE2 is not None:
+        line2 = TLE2
+
+    if num_samples is None:
+        num_samples = 100
+
+    # Extract list of times from filenames and inst_id
+    times, index, dates = ps_meth.generate_times(fnames, num_samples,
+                                                 freq=cadence)
+
+    # The observer's (ground station) position on the Earth surface
+    site = ephem.Observer()
+    site.lon = str(obs_long)
+    site.lat = str(obs_lat)
+    site.elevation = obs_alt
+
+    # The first parameter in readtle() is the satellite name
+    sat = ephem.readtle('pysat', line1, line2)
+    output_params = []
+    for timestep in index:
+        lp = {}
+        site.date = timestep
+        sat.compute(site)
+        # parameters relative to the ground station
+        lp['obs_sat_az_angle'] = ephem.degrees(sat.az)
+        lp['obs_sat_el_angle'] = ephem.degrees(sat.alt)
+        # total distance away
+        lp['obs_sat_slant_range'] = sat.range
+        # satellite location
+        # sub latitude point
+        lp['glat'] = np.degrees(sat.sublat)
+        # sublongitude point
+        lp['glong'] = np.degrees(sat.sublong)
+        # elevation of sat in m, stored as km
+        lp['alt'] = sat.elevation / 1000.
+        # get ECEF position of satellite
+        lp['x'], lp['y'], lp['z'] = OMMBV.geodetic_to_ecef(lp['glat'],
+                                                           lp['glong'],
+                                                           lp['alt'])
+        output_params.append(lp)
+
+    output = pds.DataFrame(output_params, index=index)
+    # Modify input object to include calculated parameters
+    # Put data into DataFrame
+    data = pds.DataFrame({'glong': output['glong'],
+                          'glat': output['glat'],
+                          'alt': output['alt'],
+                          'position_ecef_x': output['x'],
+                          'position_ecef_y': output['y'],
+                          'position_ecef_z': output['z'],
+                          'obs_sat_az_angle': output['obs_sat_az_angle'],
+                          'obs_sat_el_angle': output['obs_sat_el_angle'],
+                          'obs_sat_slant_range':
+                          output['obs_sat_slant_range']},
+                         index=index)
+    data.index.name = 'Epoch'
+
+    return data, meta.copy()
+
+
+list_files = functools.partial(ps_meth.list_files, test_dates=_test_dates)
+download = functools.partial(ps_meth.download)
+
+# create metadata corresponding to variables in load routine just above
+# made once here rather than regenerate every load call
```

Comment:
```suggestion
# Create metadata corresponding to variables in load routine just above.
# Defined here here rather than above to avoid regeneration at every load call.
```